homeassistant/components/group/climate.py

```diff
@@ -0,0 +1,396 @@
+    CONF_UNIQUE_ID,
+    STATE_UNAVAILABLE,
+    STATE_UNKNOWN,
+)
+from homeassistant.core import HomeAssistant, State, callback
+from homeassistant.helpers import config_validation as cv, entity_registry as er
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+
+from .entity import GroupEntity
+from .util import (
+    find_state_attributes,
+    most_frequent_attribute,
+    reduce_attribute,
+    states_equal,
+)
+
+DEFAULT_NAME = "Climate Group"
+
+# No limit on parallel updates to enable a group calling another group
+PARALLEL_UPDATES = 0
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_ENTITIES): cv.entities_domain(DOMAIN),
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_UNIQUE_ID): cv.string,
+        vol.Optional(CONF_TEMPERATURE_UNIT): cv.temperature_unit,
+    }
+)
+# edit the supported_flags
+SUPPORT_FLAGS = (
+    ClimateEntityFeature.TARGET_TEMPERATURE
+    | ClimateEntityFeature.TARGET_TEMPERATURE_RANGE
+    | ClimateEntityFeature.PRESET_MODE
+    | ClimateEntityFeature.SWING_MODE
+    | ClimateEntityFeature.FAN_MODE
+)
+
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Initialize climate.group platform."""
+    async_add_entities(
+        [
+            ClimateGroup(
+                config.get(CONF_UNIQUE_ID),
+                config[CONF_NAME],
+                config[CONF_ENTITIES],
+                config.get(CONF_TEMPERATURE_UNIT, hass.config.units.temperature_unit),
+            )
+        ]
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Initialize Climate Group config entry."""
+    registry = er.async_get(hass)
+    entities = er.async_validate_entity_ids(
+        registry, config_entry.options[CONF_ENTITIES]
+    )
+
+    async_add_entities(
+        [
+            ClimateGroup(
+                config_entry.entry_id,
+                config_entry.title,
+                entities,
+                config_entry.options.get(
+                    CONF_TEMPERATURE_UNIT, hass.config.units.temperature_unit
+                ),
+            )
+        ]
+    )
+
+
+class ClimateGroup(GroupEntity, ClimateEntity):
+    """Representation of a climate group."""
+
+    _attr_available: bool = False
+    _attr_assumed_state: bool = True
+
+    def __init__(
+        self,
+        unique_id: str | None,
+        name: str,
+        entity_ids: list[str],
+        temperature_unit: str,
+    ) -> None:
+        """Initialize a climate group."""
+        self._name = name
+        self._entity_ids = entity_ids
+
+        self._features: dict[ClimateEntityFeature, set[str]] = {
```

Comment:
Because as other group implementations, this dictionary is holding the entities in the groups that support each features or service, and then use to just call the service on top of the entities that support the feature (rather than call the service on top of all entities in the group).